tests/test_event_detection.py

```diff
@@ -1,4 +1,120 @@
+            id_val + '.mat')
+        raw_data = loadmat(fname)
+        data = raw_data['prestim_raw_yes_no']
+        # calculate TFR
+        tfrs = se.tfr(data, freqs, samp_freq)
+
+        # find events
+        subj_evs = se.find_events(tfr=tfrs, times=times, freqs=freqs,
+                                  event_band=event_band, thresholds=None,
+                                  threshold_FOM=thresh_fom)
+        py_ev_dict.append(subj_evs)
+
+    # take ev counts and timing latencies from respective dictionaries
+    # and structures to put into a subj x trial matrix for comparison
+    py_ev_count_mat = np.zeros((10, 200))
+    for subj_idx, id_val in enumerate(subj_ids):
+        py_ev_count_mat[subj_idx, :] = np.array(
+            [len(trial) for trial in py_ev_dict[subj_idx]])
+
+    # matlab - these are already in an array, so just reshape
+    matlab_ev_count_mat = np.array(
+        [trial.shape[1] for trial in matlab_ev_struct['event_times'][:, 0]])
+    matlab_ev_count_mat = np.reshape(matlab_ev_count_mat, (10, 200))
+
+    # Overall check that same # of evs detected
+    # (or a tolerable # of differences)
+    assert np.mean(np.abs(
+        matlab_ev_count_mat-py_ev_count_mat
+        )) <= dev_count_avg, "Should be mostly same number of evs"
+
+    # Once assured number of evs are within tolerable limits, extract evs
+    # matlab evs
+    same_ev_count_bool = (matlab_ev_count_mat == py_ev_count_mat)
+    same_ev_count_bool_reshape = np.reshape(same_ev_count_bool, (2000,))
+
+    # get event timing in trials in which the event count was equal
+    matlab_ev_timing_list = list()
+    py_ev_timing_list = list()
+    # extract latencies from matlab structure, which has all subject trials
+    # in array of arrays within dictionary key
+    for trial in range(len(same_ev_count_bool_reshape)):
+        if same_ev_count_bool_reshape[trial]:
+            matlab_ev_timing_list = [np.append(matlab_ev_timing_list, np.array(
+                event)) for event in matlab_ev_struct[
+                'event_times'][trial, 0]][0]
+    # extract latencies from py dictionaries for each subject
+    for subj_idx in range(10):
+        subj_evs = py_ev_dict[subj_idx]
+        for trial_idx in range(200):
+            trial_evs = list()
+            if same_ev_count_bool[subj_idx, trial_idx]:
+                # append sorted event latencies
+                if len(subj_evs[trial_idx]) > 0:
+                    for event in subj_evs[trial_idx]:
+                        trial_evs = np.append(trial_evs, event['Peak Time'])
+                    sorted = np.argsort(trial_evs)
+                    py_ev_timing_list = np.append(py_ev_timing_list,
+                                                  trial_evs[sorted])
+
+    # ensure that, in the events that were the same detected in each method,
+    # timing is within a tolerable limit of difference
+    assert ((matlab_ev_timing_list - py_ev_timing_list)*1000 <= dev_max_time
```

Comment:
```suggestion
    assert ((matlab_ev_timing_list - py_ev_timing_list) * 1000 <= dev_max_time
```
PEP8 styling. Not sure why Flake8 didn't catch this.

Reply:
Also, isn't this supposed to be the absolute value of the difference between the two times?

Reply:
I was wrong about the PEP8 styling suggestion here, so scrap my initial suggestion. Using [whitespace around lower priority operators in a sequence of operations](https://peps.python.org/pep-0008/#other-recommendations) is at the discretion of the coder, and in this case your style makes sense.